compaction/compaction_manager.cc

```diff
@@ -1477,6 +1477,10 @@ const std::vector<sstables::compaction_info> compaction_manager::get_compactions
             }) | boost::adaptors::transformed(to_info));
 }
 
+bool compaction_manager::compaction_disabled(replica::table* t) const {
+    return _compaction_state.contains(&t->as_table_state()) && _compaction_state.at(&t->as_table_state()).compaction_disabled();
```

Comment:
It's a waste to search the table state twice.
How about:
```c++
    auto it = _compaction_state.find(&t->as_table_state);
    return it != _compaction_state.end() && it->second.compaction_disabled();
```

Reply:
I think we can do it after the series. As this series is very long, I decided to only proceed with straightforward switches, without much code change.